lib/elixir/lib/code.ex

```diff
@@ -1197,6 +1197,105 @@ defmodule Code do
     :elixir.string_to_quoted!(to_charlist(string), line, column, file, opts)
   end
 
+  @doc """
+  Converts the given string to its quoted form and a list of comments.
+
+  This function is useful when performing textual changes to the source code,
+  while preserving information like comments and literals position.
+
+  Returns `{:ok, quoted_form, comments}` if it succeeds,
+  `{:error, {line, error, token}}` otherwise.
+
+  Comments are maps containing information about the line they were found, its
+  contents, and how many end of lines were found between the comment and the
+  closest tokens:
+      iex> Code.string_to_quoted_with_comments("\""
+      ...> :foo
+      ...>
+      ...> # Hello, world!
+      ...>
+      ...>
+      ...> # Some more comments!
+      ...> "\"")
+      {:ok, :foo, [
+        %{line: 3, previous_eol: 2, next_eol: 3, text: "\# Hello, world!"},
+        %{line: 6, previous_eol: 3, next_eol: 1, text: "\# Some more comments!"},
+      ]}
+
+  Check `string_to_quoted/2` for options information.
+  """
+  @spec string_to_quoted_with_comments(List.Chars.t(), keyword) ::
+          {:ok, Macro.t(), map()} | {:error, {location :: keyword, term, term}}
+  def string_to_quoted_with_comments(string, opts \\ [])
+      when is_binary(string) and is_list(opts) do
+    file = Keyword.get(opts, :file, "nofile")
+    line = Keyword.get(opts, :line, 1)
+    column = Keyword.get(opts, :column, 1)
+
+    charlist = String.to_charlist(string)
```

Comment:
You're calling `String.to_charlist/1` on this so it seems like the spec should be `String.t()`, but the rest of the functions in this module use `List.Chars.t()`. Call `to_charlist/1` instead of `String.to_charlist/1` :)

```suggestion
    charlist = to_charlist(string)
```